src/openrct2/management/Research.h

```diff
@@ -19,32 +19,80 @@ struct rct_ride_entry;
 
 struct ResearchItem
 {
-    // Bit 16 (0: scenery entry, 1: ride entry)
     union
     {
         uint32_t rawValue;
         struct
         {
-            uint8_t entryIndex;
+            ObjectEntryIndex entryIndex;
             uint8_t baseRideType;
             uint8_t type; // 0: scenery entry, 1: ride entry
-            uint8_t flags;
         };
     };
+    uint8_t flags;
     uint8_t category;
 
     bool IsNull() const;
+    void SetNull();
     bool Equals(const ResearchItem* otherItem) const;
     bool Exists() const;
     bool IsAlwaysResearched() const;
     rct_string_id GetName() const;
 
     ResearchItem() = default;
-    constexpr ResearchItem(uint32_t _rawValue, int32_t _category)
+    constexpr ResearchItem(uint32_t _rawValue, uint8_t _category, uint8_t _flags)
         : rawValue(_rawValue)
+        , flags(_flags)
         , category(_category)
     {
     }
+    ResearchItem(uint8_t _type, ObjectEntryIndex _entryIndex, uint8_t _baseRideType, uint8_t _category, uint8_t _flags)
+        : entryIndex(_entryIndex)
+        , baseRideType(_baseRideType)
+        , type(_type)
+        , flags(_flags)
+        , category(_category)
+    {
+    }
+
+    RCT12ResearchItem ToRCT12ResearchItem() const
+    {
+        RCT12ResearchItem retItem = {};
```

Comment:
```suggestion
        RCT12ResearchItem retItem {};
```

Reply:
Is this is a style thing, or does it make a difference in a technical way?

Reply:
Theoretically you're creating an empty object and assigning versus just initializing to empty, but the compiler should optimize it anyway, so it ends up being a style choice